### ISSUE TITLE
Is there an ability to write python scripts for Razer devices?

### ISSUE THREAD

**Issue by nathanngabc** [1kc/razer-macos#70]
In the open razer project, there is an ability to program your keyboard using their api via python scripting.  Is this feature available in the mac port as well?  Here are the provided examples of the scripts: https://github.com/openrazer/openrazer/tree/master/examples

Thanks in advance!

**Comment by 1kc** [comment]
Hi Nanthan, currently you can call the openrazer api in this app with Javascript only. Python scripting is not yet available, but I think it is possible to implement.

Keep in mind the apis in this project are only a subset of openrazer's, as they had to be manually rewriten on Mac.

**Comment by nathanngabc** [comment]
Hi, is there any page where I can find the documentation?

**Comment by 1kc** [comment]
Its not super well documented but I can point you to the right direction.
At the bottom of https://github.com/1kc/razer-macos/blob/master/src/driver/addon.cc , you will see a list of functions that you can call from Javascript. These functions will in turn call the C driver functions for you.

For example:

```javascript
import addon from '../driver'
addon.kbdSetModeStatic(new Uint8Array([0xff,0,0]))} // Set entire keyboard to red
```

We are not at the stage where we can set colours for individual keys yet. Maybe that is something you could look into porting over if you are interested?

**Comment by nathanngabc** [comment]
Yes, I am interested.  Could you please give some explanation on your porting process and also maybe documentation for the original code?

**Comment by 1kc** [comment]
Here is a simple explanation here regarding the porting the driver code, https://github.com/1kc/razer-macos/wiki.

The process goes:

1. Copying C functions across from https://github.com/openrazer/openrazer/blob/master/driver/razerkbd_driver.c, changing function signatures and variables accordingly. (See existing examples in https://github.com/1kc/razer-macos/blob/master/src/driver/razerkbd_driver.c)
2. Add C++ wrappers to call these C functions in https://github.com/1kc/razer-macos/blob/master/src/driver/addon.cc, this is a [Node Native Addon](https://nodejs.org/api/n-api.html)
3. Call addon functions in Javascript when user intereacts with the UI (Electron and React, razer-macos/src/main/index.js)

**Comment by nathanngabc** [comment]
How do I set everything up to begin testing?
